src/range_proof.rs

```diff
@@ -0,0 +1,558 @@
+    transcript_protocol::TranscriptProtocol,
+    utils::{bit_vector_of_scalars, div_floor_usize, nonce},
+};
+
+/// Contains the public range proof parameters intended for a verifier
+#[derive(Clone, Debug)]
+pub struct RangeProof {
+    a: CompressedRistretto,
+    a1: CompressedRistretto,
+    b: CompressedRistretto,
+    r1: Scalar,
+    s1: Scalar,
+    d1: Scalar,
+    li: Vec<CompressedRistretto>,
+    ri: Vec<CompressedRistretto>,
+}
+
+impl RangeProof {
+    /// The maximum bit length that proofs can be generated for
+    pub const MAX_BIT_LENGTH: usize = 64;
+
+    fn get_a(&self) -> Result<RistrettoPoint, ProofError> {
+        self.a.decompress().ok_or_else(|| {
+            ProofError::InternalDataInconsistent("Member 'a' was not the canonical encoding of a point".to_string())
+        })
+    }
+
+    fn get_a1(&self) -> Result<RistrettoPoint, ProofError> {
+        self.a1.decompress().ok_or_else(|| {
+            ProofError::InternalDataInconsistent("Member 'a1' was not the canonical encoding of a point".to_string())
+        })
+    }
+
+    fn get_b(&self) -> Result<RistrettoPoint, ProofError> {
+        self.b.decompress().ok_or_else(|| {
+            ProofError::InternalDataInconsistent("Member 'b' was not the canonical encoding of a point".to_string())
+        })
+    }
+
+    fn get_r1(&self) -> Scalar {
+        self.r1
+    }
+
+    fn get_s1(&self) -> Scalar {
+        self.s1
+    }
+
+    fn get_d1(&self) -> Scalar {
+        self.d1
+    }
+
+    fn get_li(&self) -> Result<Vec<RistrettoPoint>, ProofError> {
+        if !self.li.is_empty() {
+            let mut li = Vec::with_capacity(self.li.len());
+            for item in self.li.clone() {
+                li.push(item.decompress().ok_or_else(|| {
+                    ProofError::InternalDataInconsistent(
+                        "An item in member 'L' was not the canonical encoding of a point".to_string(),
+                    )
+                })?)
+            }
+            Ok(li)
+        } else {
+            Err(ProofError::InternalDataInconsistent(
+                "Vector 'L' not assigned yet".to_string(),
+            ))
+        }
+    }
+
+    fn get_ri(&self) -> Result<Vec<RistrettoPoint>, ProofError> {
+        if !self.li.is_empty() {
+            let mut ri = Vec::with_capacity(self.ri.len());
+            for item in self.ri.clone() {
+                ri.push(item.decompress().ok_or_else(|| {
+                    ProofError::InternalDataInconsistent(
+                        "An item in member 'R' was not the canonical encoding of a point".to_string(),
+                    )
+                })?)
+            }
+            Ok(ri)
+        } else {
+            Err(ProofError::InternalDataInconsistent(
+                "Vector 'R' not assigned yet".to_string(),
+            ))
+        }
+    }
+
+    /// Create a single or aggregated range proof for a single party that knows all the secrets
+    pub fn prove(
```

Comment:
Maybe add a docstring describing how the transcript should be created. Should it be a new instance? Under what circumstances can an existing Transcript be used?

Reply:
Thanks for the chat, going to hide the transcript inside the prove method